postgres/datadog_checks/postgres/statements.py

```diff
@@ -181,6 +194,39 @@ def _get_pg_stat_statements_columns(self):
                 self._stat_column_cache = col_names
                 return col_names
 
+    def _check_called_queries(self):
+        pgss_view_without_query_text = self._config.pg_stat_statements_view
+        if pgss_view_without_query_text == "pg_stat_statements":
+            # Passing false for the showtext argument leads to a huge performance increase. This
+            # allows the engine to avoid retrieving the potentially large amount of text data.
+            # The query count query does not depend on the statement text, so it's safe for this use case.
+            # For more info: https://www.postgresql.org/docs/current/pgstatstatements.html#PGSTATSTATEMENTS-FUNCS
+            pgss_view_without_query_text = "pg_stat_statements(false)"
+
+        with self._check._get_main_db() as conn:
+            with conn.cursor(cursor_factory=CommenterCursor) as cursor:
+                called_queryids = []
+                query = QUERY_COUNTS_QUERY.format(pg_stat_statements_view=pgss_view_without_query_text)
+                rows = self._execute_query(cursor, query, params=(self._config.dbname,))
+                for row in rows:
+                    queryid = row[0]
+                    if queryid is None:
+                        continue
+
+                    calls = row[1]
+                    calls_changed = self._query_count_cache.set_calls(queryid, calls)
+                    if calls_changed:
+                        called_queryids.append(queryid)
```

Comment:
Unless I miss something, you could just replace the queryid->calls map at the end.
```
# Build new queryid to calls map
queryid_to_calls = {}
for row in rows:
    queryid, calls = row
    queryid_to_calls[queryid] = calls
    if self.previous_queryid_to_calls.get(queryid, 0) != calls:
        # There's a different number of calls, keep track of the queryid
        called_queryids.append(queryid)
```
That would remove the issue of growing elements in the help.

Reply:
I couldn't use this snippet exactly, because I introduced a separate class that maintains the cache logic. So I introduced a `end_query_call_snapshot` method to that class which handles overwriting to a new cache variable.

Reply:
My point was more that we may not need the TTL cache logic (I though I've written it in my initial message but somehow it got lost...) and keeping the queryId to calls map from the previous run could be enough.
Since we completely replace the map at the end of a check run, it will always be capped to `pg_stat_statements.max` elements.

Reply:
I commented before pushing the commit I was working on - the latest code now has this logic.